src/sentry/buffer/redis.py

```diff
@@ -208,6 +208,56 @@ def get(
             col: (int(results[i]) if results[i] is not None else 0) for i, col in enumerate(columns)
         }
 
+    def get_redis_connection(self, key: str) -> Any | None:
+        # TODO type RedisPipeline instead of using Any here
+        if is_instance_redis_cluster(self.cluster, self.is_redis_cluster):
+            conn = self.cluster
+        elif is_instance_rb_cluster(self.cluster, self.is_redis_cluster):
+            conn = self.cluster.get_local_client_for_key(key)
+        else:
+            raise AssertionError("unreachable")
+
+        pipe = conn.pipeline()
+        return pipe
+
+    def push_to_set(self, key: str, value: list[int, int] | int) -> None:
+        pending_key = self._make_pending_key_from_key(key)
+        pipe = self.get_redis_connection(pending_key)
+        pipe.sadd(key, value)
```

Comment:
I'm not totally clear on whether I should be using the `key` or `pending_key` here, I think it's `key` but I'm not sure what `pending_key` is for.